src/data/DataElementsD2Repository.ts

```diff
@@ -1,23 +1,31 @@
 import _ from "lodash";
-import { D2Api, Id } from "types/d2-api";
+import { D2Api, Id, MetadataPick } from "types/d2-api";
 import { NamedRef } from "domain/entities/Base";
 import { DataElementsRepository } from "domain/repositories/DataElementsRepository";
+import { DataElement } from "domain/entities/DataElement";
 
 export class DataElementsD2Repository implements DataElementsRepository {
     constructor(private api: D2Api) {}
 
+    async getByIds(ids: Id[]): Promise<DataElement[]> {
+        return this.getDataElements(ids);
+    }
+
     async getDataElementsNames(ids: Id[]): Promise<NamedRef[]> {
+        return this.getDataElements(ids).then(dataElements =>
+            dataElements.map(de => ({ id: de.id, name: de.name }))
+        );
+    }
+
+    private async getDataElements(ids: Id[]): Promise<D2DataElement[]> {
         const metadata$ = this.api.metadata.get({
             dataElements: {
-                fields: {
-                    id: true,
-                    name: true,
-                },
+                fields: dataElementFields,
                 filter: { id: { in: ids } },
             },
         });
 
-        const { dataElements } = await metadata$.getData();
+        const dataElements = (await metadata$.getData()).dataElements;
```

Comment:
why that change? the original looks more declarative to me

Reply:
It is indeed to me also. I made the change while testing because it was more comfortable for me but I forgot to reverse it